frontend/src/components/common/HashTag/HashTag.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+
+import * as S from '@/components/common/HashTag/HashTag.styles';
+import PageLayout from '@/components/layout/PageLayout/PageLayout';
+import { validateHashTagInput } from '@/utils/validateInput';
+
+export interface HashTagProps {
+	hashTags: string[];
+	setHashTags: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+const HashTag = ({ hashTags, setHashTags }: HashTagProps) => {
+	const [hashTagInput, setHashInput] = useState<string>('');
+	const [errorMsg, setErrorMsg] = useState<string>('');
+
+	const checkIsDuplicateHashTag = () => hashTags.includes(hashTagInput);
+
+	const onHashTagEnterEventHandler = (e: React.ChangeEvent<HTMLFormElement>) => {
+		e.preventDefault();
+		if (hashTags.length === 5) {
+			setErrorMsg('해시태그는 최대 5개까지 등록이 가능합니다');
+			setHashInput('');
+			return;
+		}
+		if (!validateHashTagInput(hashTagInput)) {
+			setErrorMsg('2글자 이상 20이하여야 합니다');
+			setHashInput('');
+			return;
+		}
+		if (checkIsDuplicateHashTag()) {
+			setErrorMsg('중복된 해시태그는 입력하실 수 없습니다');
+			setHashInput('');
+			return;
+		}
+		setHashTags([...hashTags, hashTagInput]);
+		setErrorMsg('');
+		setHashInput('');
+	};
+
+	const onHashTagDeleteEventHandler = (e: React.KeyboardEvent<HTMLInputElement>) => {
+		if (e.code === 'Backspace' && hashTags.length >= 1 && hashTagInput.length === 0) {
+			if (hashTags.length <= 1) {
+				setHashTags([]);
+				return;
+			}
+
+			setHashTags(
+				hashTags.filter((item, index) => {
+					if (index < hashTags.length - 1) {
+						return item;
+					}
+				}),
+			);
```

Comment:
filter 같은 경우 true를 return 하면 그 아이템을 거르는거 아닌가요..?
```suggestion
			setHashTags(
				hashTags.filter((item, index) => index < hashTags.length -1),
			);
```

이런식으로 작성해도 되지않을까요?

Reply:
리뷰어님 읽기 편하라고 if문으로 해서 명시적으로 해놓긴 했는데
수정하겠습니다! 
뭔가 개인적으로는 인덱스로 처리하다 보니깐 바로 return 해버리면 이상하더라고요 ㅋㅋㅋㅋㅋ
그래서 일단 if문으로 넣긴 했었는데 수정하겠습니다~